twitterscraper/tweet.py

```diff
@@ -70,7 +62,12 @@ def from_soup(cls, tweet):
             .find('p', 'tweet-text')
         text_html = str(soup_html) or ""
         text = soup_html.text or ""
-        links, hashtags = cls.extract_links_and_hashtags(soup_html)
+        links = [
+            atag.get('data-expanded-url', atag['href'])
+            for atag in soup_html.find_all('a', class_='twitter-timeline-link')
+            if 'pic.twitter' not in atag.text  # eliminate picture
+        ]
+        hashtags = [tag.strip('#')for tag in re.findall(r'#\w+', text)]
```

Comment:
May I suggest to remove the `#` already in the regex? This can be accomplished with a [lookbehind](https://www.rexegg.com/regex-lookarounds.html):

```Python
hashtags = re.findall(r'(?<=#)\w+', text)
```

Reply:
@twollnik Although it looks better if its done in a single expression, the method you suggested is significantly slower.
```
t0 = time.time()
for ii in range(0,100000):
    hashtags = [tag.strip('#') for tag in re.findall(r'#\w+', text)]
t1 = time.time()
print(t1-t0)

t0 = time.time()
for ii in range(0,100000):
    hashtags = re.findall(r'(?<=#)\w+', text)
t1 = time.time()
print(t1-t0)
>>>0.885812520980835
>>>7.5316572189331055
```

Besides that, I am wondering whether we should remove the pound sign at all, since it becomes less clear it is about hashtags.

Reply:
I hadn't thought of runtime efficiency, thanks for conducting the experiment :)

Reply:
I hadn't thought efficiency, too.
Thanks for your experiment.